new/language/patternquery/builder/builder.go

```diff
@@ -107,13 +109,53 @@ func (builder *builder) build(rootNode *tree.Node) (*Result, error) {
 	paramToVariable, equalParams := builder.processVariableToParams()
 
 	return &Result{
-		Query:           builder.stringBuilder.String(),
-		ParamToVariable: paramToVariable,
-		EqualParams:     equalParams,
-		ParamToContent:  builder.paramToContent,
+		Query:             builder.stringBuilder.String(),
+		ParamToVariable:   paramToVariable,
+		EqualParams:       equalParams,
+		ParamToContent:    builder.paramToContent,
+		SkipRulesForNodes: builder.skipRulesForNodes(rootNode),
 	}, nil
 }
 
+func (builder *builder) skipRulesForNodes(rootNode *tree.Node) map[string][]*tree.Node {
+	var res map[string][]*tree.Node
+	var rulesSkippedForNode []string
+	// walk tree
+	err := rootNode.Walk(func(node *tree.Node, visitChildren func() error) error {
+		log.Error().Msgf("node type %s", node.Type())
+		if node.Type() == "comment" {
```

Comment:
Think I'm doing something wrong here. It seems none of the nodes are "comment" types. Maybe `Walk()` isn't what we want here?